### PR TITLE
added cosmo (for critical surface mass density)

### LINKED ISSUE #10: Critical surface mass density
Lensing quantity, required for GGL.

### PR DESCRIPTION
**Pull request by martinkilbinger**
cosmo.py with calculations of Sigma_crit (critical surface mass density),
and its average over source redshifts.

### CODE REVIEW

**Comment by martinkilbinger** [comment]
I added the intersphinx doc link for astropy, it is working now (e.g. astropy.units.Quantity).

I could however not find the corresponding API doc for pyccl.

File: cs_util/cosmo.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
+"""COSMO
```
Comment (sfarrens) [inline cs_util/cosmo.py]:
```suggestion
"""COSMO.
```

File: cs_util/tests/test_cosmo.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+            d_lens=self._d_lens,
+            d_source_arr=self._d_source_arr,
+        )
+        npt.assert_almost_equal(
+            sigma_crit_eff.value,
+            self._sigma_crit_value_eff,
+            decimal=3,
+        )
+
+
```
Comment (sfarrens) [inline cs_util/tests/test_cosmo.py:192]:
```suggestion
```

**Comment by sfarrens** [comment]
Hey @martinkilbinger no major comments here. I went through the implementation fairly casually, but since you have fairly extensive unit tests it should be OK. If you want to take it to the next level I would recommend adding paper references for where the specific implementation was sourced and possibly LaTeX equations in the `Notes` section of the dosctrings. Up to you. 🙂

**Comment by martinkilbinger** [comment]
> Hey @martinkilbinger no major comments here. I went through the implementation fairly casually, but since you have fairly extensive unit tests it should be OK. If you want to take it to the next level I would recommend adding paper references for where the specific implementation was sourced and possibly LaTeX equations in the `Notes` section of the dosctrings. Up to you. slightly_smiling_face

Good idea! I added a reference, it renders well locally, seems fine. I will then merge :-)

**Review by martinkilbinger — COMMENTED** [review]
OK

**Comment by martinkilbinger** [comment]
Hm, merging seems to be blocked. Maybe you need to approve the PR, @sfarrens ?

**Review by sfarrens — APPROVED** [review]

**Pull request merged**